src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -177,14 +189,15 @@ export async function makeUtxoEngine(
     },
 
     getDisplayPrivateSeed(): string | null {
-      return getMnemonic({ keys: walletInfo.keys, coin: engineInfo.network })
+      // Private key may be missing for watch-only wallets
+      const privateKey = asMaybe(asCurrencyPrivateKey)(sensitiveWalletInfo.keys)
+      if (privateKey == null)
```

Comment:
I think the intention was to return `null` here when unavailable, rather than throwing.